### PR TITLE
Adding an actionType property to CreditCardPage

### PR DESCRIPTION
**Pull request by prettyboymp**
Adds a new actionType property to CreditCardPage to distinguish between `add` and `update` card requests when retrieving the payment configuration.  

### Testing.
1. Open the Network tab of your browser's developer tools
2. Go to Manage Purchases page (http://calypso.localhost:3000/purchases).
3. Select a purchase paid with credit card.
4. Click on Edit Payment Method link.
5. Submit card details.
6. Check that the `request_type` parameter to `paygate-configuration` is set to `card_update`.
7. Go back to Manage Purchases page.
8. Pick a purchase paid with credits.
9. Click on Add Payment Method link.
10. Submit card details. 
11. Check that the `request_type` parameter to `paygate-configuration` is set to `card_update`.
12.  Go to /me/biling
13.  Choose the 'Add Credit Card' option within Manage Your Credit Cards.
14. Submit card details. 
15. Check that the `request_type` parameter to `paygate-configuration` is set to `card_add`.

Test live: https://calypso.live/?branch=update/distinguish-add-update-card-requests

### CODE REVIEW

**Comment by prettyboymp** [comment]
I'm not sure whether we want the requestType to be 'card_update' or 'card_add' when it comes to the page used when adding a card to a purchase that previously used credits. cc @janm6k

**Comment by prettyboymp** [comment]
I went ahead and switched the requestType to `card_update` when adding a card to an existing purchase.

File: client/me/purchases/payment/edit-card-details/index.jsx [hunk 1]
```diff
@@ -72,7 +72,8 @@ class EditCardDetails extends PurchaseCardDetails {
 					apiParams={ this.getApiParams() }
 					initialValues={ this.props.card }
 					recordFormSubmitEvent={ this.recordFormSubmitEvent }
-					successCallback={ this.successCallback } />
+					successCallback={ this.successCallback }
+					actionType="card_update" />
```
Comment (gziolo) [inline client/me/purchases/payment/edit-card-details/index.jsx]: I have one tiny stylistic request. Can we use this coding style:
```js
actionType="card_update"
```
in all other places? :)

**Comment by gziolo** [comment]
> I'm not sure whether we want the requestType to be 'card_update' or 'card_add' when it comes to the page used when adding a card to a purchase that previously used credits. cc @janm6k

Good question. This one is a bit tricky. Maybe we should send unique values for all 3 cases to have better control on the API side? :)

Changes look good! I left only one comment, which is super minor.

**Comment by janm6k** [comment]
I don't know about any reason why we would need to have separate type for updating a card for existing subscription, and adding a card for subscription that previously did not have card, so `card_update` feels like a good name for both.

**Comment by gziolo** [comment]
It needs to be rebased after #7661 was merged into master.

File: client/me/purchases/payment/edit-card-details/index.jsx (hunk 1 above)
Comment (prettyboymp) [inline client/me/purchases/payment/edit-card-details/index.jsx]: I don't know what I was thinking when I updated that first one.

**Pull request merged**